src/kirby/components/modal/modal-wrapper/modal-wrapper.component.tns.html

```diff
@@ -1,16 +1,22 @@
 <ModalStack class="modal-wrapper" dismissEnabled="true" (showingModally)="onShowingModally($event)">
+    <AbsoluteLayout *ngIf="config.closeBtnPlacement === 'outside'" horizontalAlignment="right">
+        <kirby-floating-action-button [showShadow]="false" (tap)="onModalDismiss()">
+            <kirby-icon [name]="config.closeIconName"></kirby-icon>
+        </kirby-floating-action-button>
+    </AbsoluteLayout>
     <StackLayout class="modal" orientation="vertical" verticalAlignment="bottom">
-        <GridLayout rows="auto" columns="*" class="title-bar">
-            <Label row="0" verticalAlignment="center" horizontalAlignment="{{ config.titleHorizontalAlignment }}"
-                text="{{ config.title }}"></Label>
-            <AbsoluteLayout row="0" verticalAlignment="center" horizontalAlignment="right">
-                <kirby-floating-action-button [showShadow]="false" (tap)="onModalDismiss()">
-                    <kirby-icon [name]="config.closeIconName"></kirby-icon>
-                </kirby-floating-action-button>
-            </AbsoluteLayout>
+        <GridLayout rows="auto, *" columns="*, auto" class="title-bar">
+
+            <Label class="title" row="0" col="0" verticalAlignment="center" horizontalAlignment="{{ config.titleHorizontalAlignment }}" textWrap="true"
+                    text="{{ config.title }}"></Label>
+            
+            <kirby-floating-action-button *ngIf="config.closeBtnPlacement === 'inside'" [showShadow]="false" (tap)="onModalDismiss()" row="0" col="1">
```

Comment:
Seems a bit duplicated... Can't you achieve this with same GridLayout? Placing on same row stacks the elements...

Reply:
I am not sure I understand, they are currently placed in a GridLayout, on the same row, but different columns

Reply:
fixed now 👍